# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../vdr-softdevice/yield.patch
# Copyright (C) 2009 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

pthread_yield() is not part of POSIX standard (and not supported by uClibc),
so use sched_yield() instead

  -- Michael Tross

--- softdevice-0.5.0/SoftOsd.c.orig	2008-01-26 00:55:44.000000000 +0100
+++ softdevice-0.5.0/SoftOsd.c	2008-11-02 13:26:57.000000000 +0100
@@ -303,7 +303,7 @@
 
         // give priority to the other threads
 #ifndef __APPLE__
-        pthread_yield();
+        sched_yield();
 #endif
 
         if (!active && !close)
